### ISSUE TITLE
Publications - Advanced Search

### ISSUE THREAD

**Issue by pimbongaerts** [pimbongaerts/mesophotic#4]
Suggestion for Search in Publications - open to suggestions of course! Tried to lay it out, so we can just have a single basic/advanced search box that is mostly the same whether user is logged in as editor/admin or not.

For basic search (descriptions for all fields can be found [here](http://mesophotic.org/about)):

- [x] single search text box
- [x] dropdown with checkboxes to choose whether it should search within `title`, `abstract` and `contents` of the pub(lication)s. By default all should be checked. 
- [x] Search by Author
- [x] dropdown with checkboxes to select which `article_type` should be considered in the search (categories: `scientific`, `technical` or `popular`). By default all should be checked.
- [x] convert search params to hash to avoid changing the api every time we add fields
- [x] dropdown with checkboxes for the different boolean fields associated with publications: `mesophotic`, `mce`, `original_data`, `new_species`. Only publications are considered that are `true` for those selected fields. By default none should be checked.

Advanced search could be in collapsible panel (as it is now), that would expand the options to include (and are already checked so always included in basic search query):

- [x] dropdown with checkboxes to select which `article_format` should be considered in the search (categories: `article`, `review`, `chapter`, `book`, `report`, `thesis`. By default all should be checked.
- [x] four dropdowns with checkboxes for the different HABTM tags that are associated with publications: `locations`, `focusgroups`, `platforms` and `fields`. By default all should be checked (and therefore should probably also have an option to uncheck everything at once)
- [x] dropdown select for the validations: `validated` (>= 3 associated, non-expired validations) or `all`. Unless logged in as editor or admin through `current_user.try(:editor_or_admin?)`, then the following options should be included as well: `expired` (expired validations for the current user), or `unvalidated` (publications with <3 associated non-expired validations and that the current user has not already validated). Default should be `all`.
- [x] make fields horizontal under search field and always visible
- [x] two select boxes (probably is a better alternative?) for the the `min_depth` and `max_depth`, allowing only integers, and by default none given so min and max_depth not evaluated.

Basically - if a search is conducted under default values, all publications should be returned.

Would be great if we could use the same query structure for the stats page - with a slightly modified search form  (so people can reduce the stats to the information they want): e.g. no text box, slightly different order and different default values.

**Comment by ryanbooker** [comment]
DONE!
